electrum/gui/qt/staking/utils.py

```diff
@@ -103,4 +103,18 @@ def get_sum_predicted_rewards(wallet: Abstract_Wallet):
             completed_period = Decimal(str(current_height - tx.staking_info.deposit_height)) / Decimal(str(tx.staking_info.staking_period))
             max_current_reward = max_reward * completed_period
             pr += max_reward * max_current_reward / tx.staking_info.accumulated_reward
+    return pr
+
+
+def get_predicted_reward(wallet: Abstract_Wallet, tx):
+    blocks_in_year = 52560  # 365 * 24 * 60 / 10
+    staking_info = wallet.network.run_from_another_thread(wallet.network.get_staking_info())
```

Comment:
cache staking info details in network object (just like feeratio) to avoid multiple calls